interlok-aws-common/src/main/java/com/adaptris/aws/STSCredentialsBuilder.java

```diff
@@ -0,0 +1,200 @@
+package com.adaptris.aws;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import org.apache.commons.lang3.ObjectUtils;
+import com.adaptris.annotation.AdvancedConfig;
+import com.adaptris.annotation.ComponentProfile;
+import com.adaptris.annotation.DisplayOrder;
+import com.adaptris.annotation.InputFieldDefault;
+import com.adaptris.annotation.InputFieldHint;
+import com.adaptris.interlok.resolver.ExternalResolver;
+import com.adaptris.security.password.Password;
+import com.adaptris.util.KeyValuePairSet;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider.Builder;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.amazonaws.services.securitytoken.model.Tag;
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+
+/**
+ * AWS credentials that makes use of AWS STS
+ * <p>
+ * AWS Security Token Service (STS) enables you to request temporary, limited-privilege credentials
+ * for AWS Identity and Access Management (IAM) users or for users that you authenticate (federated
+ * users). For more information about using this service, see
+ * <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html">Temporary
+ * Security Credentials</a>.
+ * </p>
+ *
+ * @config aws-sts-credentials-builder
+ */
+@XStreamAlias("aws-sts-credentials-builder")
```

Comment:
Can we call this something like: aws-sts-assumerole-credentials-builder? AWS uses the "assume role" terminology for this and we should reflect that since that's what people will end up searching for.

Reply:
Aaron's checkin should fix it (yeah it breaks your test config now ;) ).